src/mame/barcrest/mpu4_oki_sampled_sound.h

```diff
@@ -0,0 +1,66 @@
+// license:BSD-3-Clause
+// copyright-holders:David Haywood, James Wallace
+
+#ifndef MAME_MACHINE_MPU4_OKI_SAMPLED_SOUND_H
+#define MAME_MACHINE_MPU4_OKI_SAMPLED_SOUND_H
+
+#pragma once
+
+#include "machine/6821pia.h"
+#include "machine/6840ptm.h"
+#include "sound/okim6376.h"
+
+
+#ifdef MAME_DEBUG
+#define MPU4SSVERBOSE 1
+#else
+#define MPU4SSVERBOSE 0
+#endif
+
+#define LOG_SS(x)   do { if (MPU4SSVERBOSE) logerror x; } while (0)
+
+DECLARE_DEVICE_TYPE(MPU4_OKI_SAMPLED_SOUND, mpu4_oki_sampled_sound)
+
+class mpu4_oki_sampled_sound : public device_t, public device_mixer_interface
+{
+public:
+	// construction/destruction
+	mpu4_oki_sampled_sound(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock = 0);
+
+	void ic3_write(offs_t offset, uint8_t data);
+	uint8_t ic3_read(offs_t offset);
+
+	void ic4_write(offs_t offset, uint8_t data);
+	uint8_t ic4_read(offs_t offset);
+
+	auto cb2_handler() { return m_cb2_handler.bind(); }
+
+	void pia_gb_porta_w(uint8_t data);
+	void pia_gb_portb_w(uint8_t data);
+	uint8_t pia_gb_portb_r();
+	DECLARE_WRITE_LINE_MEMBER(pia_gb_ca2_w);
+	DECLARE_WRITE_LINE_MEMBER(pia_gb_cb2_w);
+	DECLARE_WRITE_LINE_MEMBER(output_cb2) { m_cb2_handler(state); }
```

Comment:
These should all be private, shouldn’t they?

Reply:
Yes, they shouldn't pollute the namespace. CB2 needs to leak out, as it feeds back to the mainboard in the BwB enhanced banking case, but that's what the cb2 handler is for.